src/Core/SpecificPrice/SpecificPriceFormatter.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\SpecificPrice;
+
+use PrestaShop\PrestaShop\Adapter\Product\PriceFormatter;
+
+/**
+ * This class is an extract of the ProductController->formatQuantityDiscounts() method
+ * It is still legacy code and need some heavy refactoring.
+ * This code has been extracted here for unit testing purpose
+ *
+ * @todo Refactor this class
+ */
+class SpecificPriceFormatter
+{
+    /**
+     * Calculation method to be used (tax included or not?)
+     *
+     * @var bool
+     */
+    private $isTaxIncluded;
+
+    /**
+     * Specific price data array
+     *
+     * @var array
+     */
+    private $specificPrice;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * SpecificPriceFormatter constructor.
+     *
+     * @param array $specificPrice
+     * @param bool $isTaxIncluded
+     * @param Context $context
+     */
+    public function __construct(array $specificPrice, bool $isTaxIncluded, $context)
```

Comment:
@matthieu-rolland It lacks
```suggestion
    public function __construct(array $specificPrice, bool $isTaxIncluded, Context $context)
```

Reply:
@matthieu-rolland Indeed just this modification left and we're good to go ;)

Reply:
done !